File: Guanabara - POO Java/Meus exercícios POO/Aula15-Exercicio_Final/src/main/java/projeto_youtube/Video.java
package projeto_youtube;
public class Video implements AcoesVideo { //a classe `Video` vai declarar os métodos abstratos da Interface `AcoesVideo`
    //Atributos Privados
    private String titulo;
    private int avaliacao, views, curtidas;
    private boolean reproduzindo;

    //Construtor é o método que é executado assim que o objeto for criado/instanciado
    public Video(String titulo) { //o construtor recebe o mesmo nome de sua classe
        //Único aatributo que será parâmetro na criação do objeto
        this.titulo = titulo;
        //Atributos que, por já serem pré-configurados, não serão parâmetros na criação do objeto
        this.avaliacao = 0;
        this.views = 0;
        this.curtidas = 0;
        this.reproduzindo = false;        
    }
    
    //Métodos Acessores (conceito do Polimorfismo)
    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public int getAvaliação() {
        return avaliacao;
    }

    public void setAvaliação(int avaliação) {
        int nova;
        nova = (this.avaliacao + avaliação) / this.views; //a avaliação do video será a média de avaliações
        this.avaliacao = nova;
    }

    public int getViews() {
        return views;
    }

    public void setViews(int views) {
        this.views = views;
    }

    public int getCurtidas() {
        return curtidas;
    }

    public void setCurtidas(int curtidas) {
        this.curtidas = curtidas;
    }

    public boolean isReproduzindo() {
        return reproduzindo;
    }

    public void setReproduzindo(boolean reproduzindo) {
        this.reproduzindo = reproduzindo;
    }
    
    //Implementações da classe `AcoesVideo`
    @Override  //é a anotação para indicar herança por sobreposição
    public void play() {
        this.reproduzindo = true;
    }

    @Override
    public void pause() {
        this.reproduzindo = false;
    }

    @Override
    public void like() {
        this.curtidas ++;
    }

    @Override //cria um Polimorfismo de Sobreposição para o método-automático `toString()`
    public String toString() {
        return "Video{" + "titulo=" + titulo + ", avaliacao=" + avaliacao + ", views=" + views + ", curtidas=" + curtidas + ", reproduzindo=" + reproduzindo + '}';
    }
    
}

File: PHP/exercicios-PHPOO/Aula14e15/Video.php
<?php
require_once 'AcoesVideo.php';
class Video implements AcoesVideo
{
    //Atributos
    private $titulo;
    private $avaliacoes;
    private $visualizacoes;
    private $curtidas;
    private $reproduzindo;

    //Métodos Implementados da Interface
    public function like()
    {
        $this->curtidas ++;
    }
    public function pause()
    {
        $this->reproduzindo = false;
    }
    public function play()
    {
        $this->reproduzindo = true;
    }
    
    //Método Construtor
    public function __construct($titulo)
    {
        $this->titulo = $titulo;
        $this->avaliacoes = 1;
        $this->visualizacoes = 0;
        $this->curtidas = 0;
        $this->reproduzindo = false;
    }

    //Métodos Acessores
    public function getTitulo()
    {
        return $this->titulo;
    }
    public function getAvaliacoes()
    {
        return $this->avaliacoes;
    }
    public function getVisualizacoes()
    {
        return $this->visualizacoes;
    }
    public function getCurtidas()
    {
        return $this->curtidas;
    }
    public function getReproduzindo()
    {
        return $this->reproduzindo;
    }
    public function setTitulo($titulo): void
    {
        $this->titulo = $titulo;
    }
    public function setAvaliacoes($avaliacoes): void
    {
        $this->avaliacoes = $avaliacoes;
    }
    public function setVisualizacoes($visualizacoes): void
    {
        $this->visualizacoes = $visualizacoes;
    }
    public function setCurtidas($curtidas): void
    {
        $this->curtidas = $curtidas;
    }
    public function setReproduzindo($reproduzindo): void
    {
        $this->reproduzindo = $reproduzindo;
    }
}
